### ISSUE TITLE
Extensions frequently freeze and need to be restarted to function properly／擴充經常假死，要重啓擴充才能工作

### ISSUE THREAD

**Issue by YheonYeung** [AdguardTeam/AdguardBrowserExtension#3079]
### Please answer the following questions for yourself before submitting an issue

- [X] Filters were updated before reproducing an issue
- [X] I checked the [knowledge base](https://adguard.com/kb/) and found no answer
- [X] I checked to make sure that this issue has not already been filed

### AdGuard Extension version

 5.0.178

### Browser version

Chrome 131.0.6778.205 (64 bit)

### OS version

win 11

### Ad Blocking

AdGuard Base filter, AdGuard Quick Fixes filter

### Privacy

AdGuard Tracking Protection filter, AdGuard URL Tracking filter

### Annoyances

AdGuard Popups filter, AdGuard Other Annoyances filter

### Language-specific

AdGuard Chinese filter, AdGuard Japanese filter

### Issue Details

Steps to reproduce:
1.Just browsing the internet, then trying to open YouTube to watch videos, you might notice ads or some websites having layout issues that feel off
2.
3.

### Actual Behavior

The extension's options page opens blank, and the extension needs to be restarted.

### Screenshots

![image](https://github.com/user-attachments/assets/3b39179a-c197-4f54-b903-9392af707db8)

### Additional Information

[adg_ext_settings_5.0.178_271224-151330.json](https://github.com/user-attachments/files/18258331/adg_ext_settings_5.0.178_271224-151330.json)

https://github.com/user-attachments/assets/c545beef-3c4e-49cf-9a07-a44806f628b1

**Comment by windwak3r** [comment]
@YheonYeung, hello!

Is the popup of the extension blank too?
Looks like a dup. of https://github.com/AdguardTeam/AdguardBrowserExtension/issues/3059

Please reproduce the problem, open the popup of the extension, then right click on it, click `Inspect` and copy the content of the `Console` tab.
<details>
  <summary>Screenshot 1</summary>

  ![Screenshot 1](https://github.com/user-attachments/assets/51de764a-2cc3-4cc8-9883-796eec6bacd0)

</details>

Then attach the logs to a comment here.

**Comment by YheonYeung** [comment]
![image](https://github.com/user-attachments/assets/af39a22c-c174-4f98-afd2-cb432b1ca223)


![image](https://github.com/user-attachments/assets/51fae1f3-4262-4b51-87c4-c17968c29f30)
is empty


![image](https://github.com/user-attachments/assets/888dc28d-fb4e-4846-ad95-75f20e25c1b2)
options page is empty too.

**Comment by windwak3r** [comment]
@YheonYeung Does the error appear if you press `Ctrl + R` / `Cmd + R` on the `Console` tab from the instructions I previously gave you?

**Comment by Fabian42** [comment]
I am getting similar freezes in Firefox 133.0.3, extension version 4.4.49. Any useful logs or settings I can send without having to use the extension's GUI?

**Comment by Fabian42** [comment]
I disabled all extensions except AdGuard and containers, still the same issue.
There is a feature to debug extensions, but all I could get from that is this screenshot:
![image](https://github.com/user-attachments/assets/81adfa53-aa71-4928-9c3b-f0fd371d1bcc)
If I try interacting with that at all, it disappears, because I have to click somewhere other than the extension menu, so it closes itself. THAT still reacts instantly. -.-
I could not find anything relevant in any of the extension's files, so no idea where it stores its settings. No luck exporting my custom filters then, RIP…
I have created a backup of my entire Firefox profile folder now and reinstalled it, so now it works, but of course all my settings are gone. Would be nice if I could restore them somehow from that backup.

**Comment by YheonYeung** [comment]
I've been using version 5.0.183 for a few days, and I haven't observed any freezing issues.

![Image](https://github.com/user-attachments/assets/4ffcb615-410d-4ab7-bcd0-1a9950264165)

**Comment by Fabian42** [comment]
Wait, 5.x exists? Why is that not in the Firefox extensions store? Is it in beta or something? The changelog page does not indicate that.

**Comment by maximtop** [comment]
>Wait, 5.x exists? Why is that not in the Firefox extensions store? Is it in beta or something? The changelog page does not indicate that.

It only exists for Chrome on MV3.

**Comment by windwak3r** [comment]
@Fabian42, hello!

> There is a feature to debug extensions, but all I could get from that is this screenshot

Can you please save all content of the Console to a file and attach it to a comment here?

Do you have exact steps to reproduce the problem?

> I have created a backup of my entire Firefox profile folder now and reinstalled it, so now it works, but of course all my settings are gone. Would be nice if I could restore them somehow from that backup.

You can use this in the Console to see your settings:
`chrome.storage.local.get(null, console.log)`

**Comment by Fabian42** [comment]
> Can you please save all content of the Console to a file and attach it to a comment here?

No, because when I click on that window at all, it empties, because the extension popup closes.

> Do you have exact steps to reproduce the problem?

I just opened the extension menu/popup while in the bugged state. No idea how to get into that state. And the only way to get out of it was to reset the extension completely.

> You can use this in the Console to see your settings: `chrome.storage.local.get(null, console.log)`

You mean the one where I sent the screenshot? That does not give me my custom filters. It mostly outputs a lot of translation strings.

BTW, I found an old backup of my custom filters in the meantime. That is still months back, but at least something.

**Comment by windwak3r** [comment]
@Fabian42 @YheonYeung We'll close the issue for now, since it's no longer recurring and there are no steps to reproduce it. Please leave a comment if the problem is still relevant.
